plugins/outputs/elasticsearch/README.md

```diff
@@ -323,6 +323,16 @@ to use them.
   ## no pipeline is used for the metric.
   # use_pipeline = "{{es_pipeline}}"
   # default_pipeline = "my_pipeline"
+
+  ## Template Index Settings
+  ## Overrides the template settings.index section with any provided options.
+  ## Defaults provided here in the config
+  # template_index_settings = {
+  #   refresh_interval = "10s",
+  #   mapping.total_fields.limit = 5000,
+  #   auto_expand_replicas = "0-1",
+  #   codec = "best_compression"
+  # }
```

Comment:
Did you find a list of these options or settings? I'm curious if we should build a struct to contain the options or just leave this wide open for the user.

Reply:
There are a lot, we could build a struct for it but it would be quite large I think. There's top level settings [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-modules-settings) and then many other subcategories of settings as well

Reply:
Yeah that's a lot :\ I'll resolve this question. Thanks!